build_tools/upload_logs_to_s3.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+"""
+upload_logs_to_s3.py
+
+Uploads build logs and index.html to an S3 bucket using the AWS CLI via subprocess.
+Intended for use in CI (e.g., GitHub Actions) at the end of a build, regardless of success/failure.
+"""
+
+import os
+import glob
+import subprocess
+import sys
+
+
+def log(*args, **kwargs):
+    # Wrapper for print that flushes immediately.
+    print(*args, **kwargs)
+    sys.stdout.flush()
+
+
+def run_aws_cp(src, dest, content_type=None):
```

Comment:
When known, prefer to add type annotations to function arguments. Here it looks like the script is using `str` for `src` _and_ `dest`, but you could also use `Path` for `src`.

Reply:
Good suggestion - added annotations.